File: app/src/main/java/com/antyzero/smoksmog/ui/screen/settings/SettingsActivity.java
package com.antyzero.smoksmog.ui.screen.settings;


import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.Toolbar;

import com.antyzero.smoksmog.R;
import com.antyzero.smoksmog.ui.BaseActivity;

import butterknife.Bind;

public class SettingsActivity extends BaseActivity {

    @Bind( R.id.toolbar )
    Toolbar toolbar;

    @Override
    protected void onCreate( Bundle savedInstanceState ) {
        super.onCreate( savedInstanceState );
        setContentView( R.layout.activity_settings );

        setSupportActionBar( toolbar );

        if ( getSupportActionBar() != null ) {
            getSupportActionBar().setDisplayHomeAsUpEnabled( true );
            getSupportActionBar().setTitle( R.string.title_settings );
        }

        getFragmentManager().beginTransaction()
                .replace( R.id.contentFragment, GeneralSettingsFragment.create() )
                .commit();
    }

    /**
     * @param context for starting
     */
    public static void start( Context context ) {
        context.startActivity( intent( context ) );
    }

    @NonNull
    private static Intent intent( Context context ) {
        return new Intent( context, SettingsActivity.class );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/settings/SettingsActivity.kt
package com.antyzero.smoksmog.ui.screen.settings


import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.antyzero.smoksmog.R
import com.antyzero.smoksmog.ui.BaseActivity
import kotlinx.android.synthetic.main.activity_settings.*

class SettingsActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        setSupportActionBar(toolbar)

        if (supportActionBar != null) {
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            supportActionBar!!.setTitle(R.string.title_settings)
        }

        fragmentManager.beginTransaction()
                .replace(R.id.contentFragment, GeneralSettingsFragment.create())
                .commit()
    }

    companion object {

        /**
         * @param context for starting
         */
        fun start(context: Context) {
            context.startActivity(intent(context))
        }

        private fun intent(context: Context): Intent {
            return Intent(context, SettingsActivity::class.java)
        }
    }
}
